### PR TITLE
jt graph command

### PR DESCRIPTION
**Pull request by chrisseaton**
For quickly seeing what a graph for a method looks like. You can also `--watch` and see how the graph changes as you modify the file.

### CODE REVIEW

File: tool/jt.rb [hunk 1]
```diff
@@ -2002,6 +2003,75 @@ def profile(*args)
...
+        raise if arg.start_with?('-')
+        raise if test_file
+        test_file = arg
+      end
+    end
+
+    raise unless test_file
+
+    if Gem::Specification.find_all_by_name('seafoam').empty?
+      sh env, 'gem', 'install', 'seafoam'
```
Comment (eregon) [inline tool/jt.rb:2029]: `env` doesn't seem defined here.
`gem install` might also be rather slow, how about checking if there is a sibling `seafoam` repo or clone it if not there?

File: tool/jt.rb [hunk 2]
```diff
@@ -2002,6 +2003,75 @@ def profile(*args)
...
+          if line =~ /\[engine\] opt done     #{method}/
+            puts line
+            Process.kill 'INT', pipe.pid
+            break
+          end
+        end
+      end
+
+      dumps = Dir.glob('graal_dumps/*').sort.last
+      graal = Dir.glob("#{dumps}/*[#{method}].bgv").first
```
Comment (eregon) [inline tool/jt.rb:2053]: Should we pick the last compilation rather if there are multiple compilations?

Comment (eregon) [reply tool/jt.rb:2053]:
```suggestion
      graph = Dir.glob("#{dumps}/*[#{method}].bgv").first
```

**Review by eregon — COMMENTED** [review]
:+1: looks convenient :)
I've been thinking at something like that but never got to script it.

**Comment by chrisseaton** [comment]
I'm working on a whole web service to do it interactively, like Godbolt but for Ruby https://github.com/ruby-compiler-survey/ruby-compiler-explorer.

File: tool/jt.rb (hunk 1 above)
Comment (chrisseaton) [reply tool/jt.rb:2029]: I was copying Rubocop which is installed as a gem. I can clone if you think it's important. I'll fix `env` either way.

**Review by wildmaples — APPROVED** [review]
🎉 I'll be using this

File: tool/jt.rb [hunk 3]
```diff
@@ -2002,6 +2003,75 @@ def profile(*args)
     app_open svg_filename
   end
 
+  def graph(*args)
+    test_file = nil
+    method = 'Object#foo'
```
Comment (wildmaples) [inline tool/jt.rb:2008]: Do you mean to hardcode the method foo? I thought user can specify the method e.g. `jt graph --method fib test.rb`

Comment (chrisseaton) [reply tool/jt.rb:2009]: This is a default - it's overwritten below.

File: tool/jt.rb (hunk 1 above)
Comment (eregon) [reply tool/jt.rb:2030]: Ah, seafoam is run on the Ruby running JT, typically CRuby, so then it doesn't matter as it'll need to be installed only once in that Ruby

Comment (eregon) [reply tool/jt.rb:2030]: It'd be good to set `env = { 'PATH' => "#{File.dirname(ruby)}:#{ENV['PATH']}" }` like for RuboCop, so we make sure the same Ruby running jt is the one running seafoam, and not potentially some other ruby in PATH which `Gem::Specification.find_all_by_name('seafoam')` would not check.

Comment (eregon) [reply tool/jt.rb:2030]: I'll do it

**Review by eregon — APPROVED** [review]

File: tool/jt.rb [hunk 4]
```diff
@@ -2002,6 +2004,78 @@ def profile(*args)
...
+
+    options = [
+      '--experimental-options',
+      '--engine.TraceCompilation',
+      '--engine.BackgroundCompilation=false',
+      "--engine.CompileOnly=#{method}",
+      '--engine.MultiTier=false',
+      '--engine.NodeSourcePositions',
+      '--vm.Dgraal.PrintGraphWithSchedule=true',
+      '--vm.Dgraal.PrintBackendCFG=true',
```
Comment (eregon) [inline tool/jt.rb:2041]: Does seafoam need these two options?

**Pull request merged**